include/gauxc/molecule.hpp

```diff
@@ -7,42 +7,57 @@
  */
 #pragma once
 
-#include <gauxc/atom.hpp>
-#include <vector>
 #include <algorithm>
-
+#include <gauxc/atom.hpp>
 #include <gauxc/gauxc_config.hpp>
+#include <type_traits>
+#include <vector>
 
 namespace GauXC {
 
 class Molecule : public std::vector<Atom> {
+private:
+  /// Determines if type @p T is different from Molecule
+  template <typename T>
+  using is_not_molecule_t = std::negation<std::is_same<T, Molecule>>;
 
-public:
+  /// Determines if all types in @p Args are different from Molecule
+  template <typename... Args>
+  using all_not_molecule = std::conjunction<is_not_molecule_t<Args>...>;
 
+  /// Disables a function via SFINAE if any type in @p Args ia Molecule
   template <typename... Args>
-  Molecule( Args&&... args ) :
-    std::vector<Atom>( std::forward<Args>(args)... ) { }
+  using disable_if_molecule_t =
+      std::enable_if_t<all_not_molecule<Args...>::value>;
+
+public:
+  template <typename... Args,
+            typename = disable_if_molecule_t<std::decay_t<Args>...>>
+  Molecule(Args &&...args) : std::vector<Atom>(std::forward<Args>(args)...) {}
 
-  Molecule( const Molecule& )          = default;
```

Comment:
Please remove linting artifacts. I won't point them out everywhere, but there's quite a few.

Reply:
That's what I get for not looking at the diffs before submitting. Will do! I probably won't get to it until tomorrow morning.

Reply:
Fixed.